### PR TITLE
ci: Add basic ci build

### PR DESCRIPTION
**Pull request by pointbazaar**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module github.com/9elements/bmc-test-go
 
-go 1.23.1
+go 1.23
```
Comment (AtomicFS) [inline go.mod:3]: Why changing this?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
-bmc-test-go
+bmc-test-cross
+bmc-test-native
```
Comment (AtomicFS) [inline .gitignore:2]: Call me nitpick but could we use format `<project-name>-<arch>`? Meaning `bmc-test-go-x86` or `bmc-test-go-arm64` or `bmc-test-go-armv5`?

There are plenty of ARM architectures and they are not always compatible with each other (32b vs 64b, etc).

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  merge_group:
+  push:
+    branches: ['main']
+    tags: ['v*']
+
+permissions:
+  contents: read
+
+jobs:
+  BuildCross:
```
Comment (AtomicFS) [inline .github/workflows/ci-build.yml:15]: If possible I would like to stick with `dash-case` (also known as `kebab-case`) naming convention.

File: .github/workflows/ci-build.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: install dependencies
+        shell: bash
+        run: sudo apt-get install -y golang make
+      - name: Build
+        shell: bash
+        run: make cross
+
+  BuildNative:
```
Comment (AtomicFS) [inline .github/workflows/ci-build.yml:26]: `dash-case` pls

File: .github/workflows/ci-build.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: install dependencies
+        shell: bash
+        run: sudo apt-get install -y golang make
+      - name: Build
```
Comment (AtomicFS) [inline .github/workflows/ci-build.yml:33]: `build` instead of `Build` pls

File: go.mod (hunk 1 above)
Comment (pointbazaar) [reply go.mod:3]: the gopher said so https://github.com/9elements/bmc-test-go/actions/runs/10815516997/job/30004587016

File: .gitignore (hunk 2 above)
Comment (pointbazaar) [reply .gitignore]: Changed it to be more descriptive. Also added a symlink so we can keep the tool name the same for now.

File: .github/workflows/ci-build.yml (hunk 4 above)
Comment (pointbazaar) [reply .github/workflows/ci-build.yml]: done

File: .github/workflows/ci-build.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+  BuildCross:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: install dependencies
+        shell: bash
+        run: sudo apt-get install -y golang make
+      - name: Build
```
[the remark this answers is not in the document]
Comment (pointbazaar) [reply .github/workflows/ci-build.yml]: done

File: .github/workflows/ci-build.yml (hunk 3 above)
Comment (pointbazaar) [reply .github/workflows/ci-build.yml]: done

File: .github/workflows/ci-build.yml (hunk 5 above)
Comment (pointbazaar) [reply .github/workflows/ci-build.yml]: done

File: go.mod (hunk 1 above)
Comment (AtomicFS) [reply go.mod:3]: Fair enough

**Review by AtomicFS — APPROVED** [review]

**Pull request merged**